heron/proto/topology.proto

```diff
@@ -69,7 +75,10 @@ message Config {
     required string key = 1;
     // Only one of the below are set
     optional string value = 2;
-    optional bytes java_serialized_value = 3;
+    //optional bytes java_serialized_value = 3;
+    optional bytes serialized_value = 3;
+
+    required ConfigValueType type = 4;
```

Comment:
@billonahill Also, I added a required field to Component in PR #1149, which might have not been a great idea for the same reason. Do you think it's better to make it optional as well?